src/consensus/blockchain_check_conditions.py

```diff
@@ -89,9 +78,20 @@ def blockchain_assert_relative_time_exceeds(condition: ConditionVarPair, unspent
     return None
 
 
+def blockchain_assert_announcement(condition: ConditionVarPair, announcements: List[Announcement]) -> Optional[Err]:
+    """
+    Check if an announcement is included in the list of announcements
+    """
+    announcement_hash = condition.vars[0]
+    if announcement_hash not in [a.name() for a in announcements]:
```

Comment:
Might be worth pre-calculating as a `set` of `a.name()` items in the caller so this isn't `O(n^2)`.

Reply:
fixed